libkbfs/keybase_daemon_local.go

```diff
@@ -236,6 +250,139 @@ func (k *KeybaseDaemonLocal) Identify(
 	return k.Resolve(ctx, assertion)
 }
 
+// ResolveIdentifyImplicitTeam implements the KeybaseService interface
+// for KeybaseDaemonLocal.
+func (k *KeybaseDaemonLocal) ResolveIdentifyImplicitTeam(
+	ctx context.Context, assertions, suffix string, tlfType tlf.Type,
+	doIdentifies bool, reason string) (ImplicitTeamInfo, error) {
+	if err := checkContext(ctx); err != nil {
+		return ImplicitTeamInfo{}, err
+	}
+
+	if tlfType != tlf.Private && tlfType != tlf.Public {
+		return ImplicitTeamInfo{}, fmt.Errorf(
+			"Invalid implicit team TLF type: %s", tlfType)
+	}
+
+	k.lock.Lock()
+	defer k.lock.Unlock()
+
+	// Canonicalize the name.
+	writerNames, readerNames, _, err :=
+		splitAndNormalizeTLFName(assertions, tlfType)
+	if err != nil {
+		return ImplicitTeamInfo{}, err
+	}
+	var writers, readers []libkb.NormalizedUsername
+	var unresolvedWriters, unresolvedReaders []keybase1.SocialAssertion
+	resolve := func(
```

Comment:
It seems like the only 2 things needed for this closure are `k` and `ctx`, in which case I think it makes sense to have this be its own helper function. It's possible that the Go compiler these days is good enough to optimize out closures when it can, but I'd prefer to continue separating them out just in case.